app/uk/gov/hmrc/cdsreimbursementclaimfrontend/models/reimbursement/DutyType.scala

```diff
@@ -0,0 +1,161 @@
+    CiderPerry             -> List(NI431, NI481, NI483, NI485, NI487),
+    HydrocarbonOils        -> List(
+      NI511,
+      NI511,
+      NI520,
+      NI521,
+      NI522,
+      NI540,
+      NI541,
+      NI542,
+      NI551,
+      NI556,
+      NI561,
+      NI570,
+      NI571,
+      NI572
+    ),
+    Biofuels               -> List(NI589, NI595),
+    MiscellaneousRoadFuels -> List(NI591, NI592),
+    Tobacco                -> List(NI611, NI615, NI619, NI623, NI627, NI633),
+    ClimateChangeLevy      -> List(NI99A, NI99B, NI99C, NI99D)
+  )
+
+  def toDutyType(repr: String): Option[DutyType] =
+    repr match {
+      case "ukduty"                 => Some(UkDuty)
+      case "euduty"                 => Some(EuDuty)
+      case "beer"                   => Some(Beer)
+      case "wine"                   => Some(Wine)
+      case "madewine"               => Some(MadeWine)
+      case "lowalcoholbeverages"    => Some(LowAlcoholBeverages)
+      case "spirits"                => Some(Spirits)
+      case "ciderperry"             => Some(CiderPerry)
+      case "hydrocarbonoils"        => Some(HydrocarbonOils)
+      case "biofuels"               => Some(Biofuels)
+      case "miscellaneousroadfuels" => Some(MiscellaneousRoadFuels)
+      case "tobacco"                => Some(Tobacco)
+      case "climatechangelevy"      => Some(ClimateChangeLevy)
+      case _                        => None
+    }
+
+  def typeToString(dutyType: DutyType): String =
+    dutyType match {
+      case UkDuty                 => UkDuty.toString.toLowerCase(Locale.UK)
+      case EuDuty                 => EuDuty.toString.toLowerCase(Locale.UK)
+      case Beer                   => Beer.toString.toLowerCase(Locale.UK)
+      case Wine                   => Wine.toString.toLowerCase(Locale.UK)
+      case MadeWine               => MadeWine.toString.toLowerCase(Locale.UK)
+      case LowAlcoholBeverages    => LowAlcoholBeverages.toString.toLowerCase(Locale.UK)
+      case Spirits                => Spirits.toString.toLowerCase(Locale.UK)
+      case CiderPerry             => CiderPerry.toString.toLowerCase(Locale.UK)
+      case HydrocarbonOils        => HydrocarbonOils.toString.toLowerCase(Locale.UK)
+      case Biofuels               => Biofuels.toString.toLowerCase(Locale.UK)
+      case MiscellaneousRoadFuels => MiscellaneousRoadFuels.toString.toLowerCase(Locale.UK)
+      case Tobacco                => Tobacco.toString.toLowerCase(Locale.UK)
+      case ClimateChangeLevy      => ClimateChangeLevy.toString.toLowerCase(Locale.UK)
+    }
+
+  implicit val taxCategoryEq: Eq[DutyType] = Eq.fromUniversalEquals[DutyType]
+
+  implicit val format: OFormat[DutyType] = derived.oformat[DutyType]()
+
+  def apply(value: Int): DutyType = DutyType.dutyTypes(value)
+
+  def unapply(dutyType: DutyType): Option[Int] = {
+    val index = DutyType.dutyTypes.indexOf(dutyType)
+    if (index > -1) {
+      Some(index)
+    } else {
+      None
```

Comment:
Can be simplify with
`Option(DutyType.dutyTypes.indexOf(dutyType)).filter(_  >  -1)`

Reply:
I would argue that the if-statement is simpler :wink: but I've changed it as you suggested.